core/src/test/java/kafka/log/remote/RemoteLogManagerTest.java

```diff
@@ -1588,6 +1588,95 @@ private void doTestFindOffsetByTimestamp(long ts, long startOffset, int targetLe
+        TopicPartition tp = new TopicPartition("sample", 0);
+        TopicIdPartition tpId = new TopicIdPartition(Uuid.randomUuid(), tp);
+        Map<String, Uuid> topicIds = Collections.singletonMap(tp.topic(), tpId.topicId());
+
+        List<EpochEntry> epochEntries = new ArrayList<>();
+        epochEntries.add(new EpochEntry(0, 0L));
+        epochEntries.add(new EpochEntry(1, 20L));
+        epochEntries.add(new EpochEntry(3, 50L));
+        epochEntries.add(new EpochEntry(4, 100L));
+        epochEntries.add(new EpochEntry(5, 200L));
+        checkpoint.write(epochEntries);
+        LeaderEpochFileCache cache = new LeaderEpochFileCache(tp, checkpoint, scheduler);
+        when(mockLog.leaderEpochCache()).thenReturn(Option.apply(cache));
+
+        long timestamp = time.milliseconds();
+        RemoteLogSegmentMetadata metadata0 = new RemoteLogSegmentMetadata(new RemoteLogSegmentId(tpId, Uuid.randomUuid()),
+                0, 99, timestamp, brokerId, timestamp, 1024, Optional.empty(), RemoteLogSegmentState.COPY_SEGMENT_FINISHED, truncateAndGetLeaderEpochs(epochEntries, 0L, 99L));
+        RemoteLogSegmentMetadata metadata1 = new RemoteLogSegmentMetadata(new RemoteLogSegmentId(tpId, Uuid.randomUuid()),
+                100, 199, timestamp + 1, brokerId, timestamp + 1, 1024, Optional.empty(), RemoteLogSegmentState.COPY_SEGMENT_FINISHED, truncateAndGetLeaderEpochs(epochEntries, 100L, 199L));
+        // Note that the metadata2 is in COPY_SEGMENT_STARTED state
+        RemoteLogSegmentMetadata metadata2 = new RemoteLogSegmentMetadata(new RemoteLogSegmentId(tpId, Uuid.randomUuid()),
+                100, 299, timestamp + 2, brokerId, timestamp + 2, 1024, Optional.empty(), RemoteLogSegmentState.COPY_SEGMENT_STARTED, truncateAndGetLeaderEpochs(epochEntries, 200L, 299L));
+
+        when(remoteLogMetadataManager.listRemoteLogSegments(eq(tpId), anyInt()))
+            .thenAnswer(ans -> {
+                int epoch = ans.getArgument(1);
+                if (epoch < 4) {
+                    return Collections.singletonList(metadata0).iterator();
+                } else if (epoch == 4) {
+                    return Arrays.asList(metadata1, metadata2).iterator();
+                } else {
+                    throw new IllegalArgumentException("Unexpected call!");
+                }
+            });
+
+        FileRecords.TimestampAndOffset expectedLocalResult = new FileRecords.TimestampAndOffset(timestamp, 9999, Optional.of(Integer.MAX_VALUE));
+        FileRecords.TimestampAndOffset expectedRemoteResult = new FileRecords.TimestampAndOffset(timestamp, 999, Optional.of(Integer.MAX_VALUE));
+        Partition mockFollowerPartition = mockPartition(tpId);
+
+        LogSegment logSegment = mock(LogSegment.class);
+        mockLogSegment(logSegment, 50L, timestamp + 10, expectedLocalResult);
+        when(mockLog.logSegments()).thenReturn(Collections.singletonList(logSegment));
```

Comment:
Could we let `mockLog.logSegments()` return multiple segments, so that we can verify the local log search works as expected? Thanks.

Reply:
Updated the test to return multiple local log segments.